projects/observability/src/shared/components/cartesian/d3/chart/cartesian-chart.ts

```diff
@@ -65,14 +68,35 @@ export class DefaultCartesianChart<TData> implements CartesianChart<TData> {
     onEvent: ChartEventListener<TData>;
   }[] = [];
 
+  protected readonly brushBehaviour: BrushBehavior<unknown>;
+
   public constructor(
     protected readonly hostElement: Element,
     protected readonly injector: Injector,
     protected readonly renderingStrategy: RenderingStrategy,
     protected readonly svgUtilService: SvgUtilService,
     protected readonly d3Utils: D3UtilService,
     protected readonly domRenderer: Renderer2
-  ) {}
+  ) {
+    this.brushBehaviour = brushX<unknown>().on('end', () => this.onBrushSelection(_d3CurrentEvent));
+  }
+
+  protected onBrushSelection(event: D3BrushEvent<unknown>): void {
+    if (!event.selection) {
+      return;
+    }
+
+    const { height } = this.hostElement.getBoundingClientRect();
+
+    const [startX, endX] = event.selection as [number, number];
+
+    let fromdata: any = this.allSeriesData.flatMap(viz => viz.dataForLocation({ x: startX, y: height }));
+    let todata: any = this.allSeriesData.flatMap(viz => viz.dataForLocation({ x: endX, y: height }));
+
+    this.eventListeners.forEach(listener => {
+      listener.onEvent({ start: fromdata[0].dataPoint.timestamp, end: todata[0].dataPoint.timestamp } as any);
```

Comment:
This depends on the api being discussed above.

Reply:
done